test/e2e/targets/googlecloudstorage/main.go

```diff
@@ -0,0 +1,259 @@
+/*
+Copyright 2022 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package googlecloudstorage
+
+import (
+	"context"
+	"io/ioutil"
+	"net/url"
+
+	. "github.com/onsi/ginkgo/v2" //nolint:stylecheck
+	. "github.com/onsi/gomega"    //nolint:stylecheck
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	clientset "k8s.io/client-go/kubernetes"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/option"
+
+	cloudevents "github.com/cloudevents/sdk-go/v2"
+
+	"github.com/triggermesh/triggermesh/test/e2e/framework"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/apps"
+	e2ece "github.com/triggermesh/triggermesh/test/e2e/framework/cloudevents"
+	"github.com/triggermesh/triggermesh/test/e2e/framework/ducktypes"
+	e2egcloud "github.com/triggermesh/triggermesh/test/e2e/framework/gcloud"
+	e2estorage "github.com/triggermesh/triggermesh/test/e2e/framework/gcloud/storage"
+)
+
+/* This test suite requires:
+
+   - A Google Cloud Service Account key in JSON format, exported in the environment as GCLOUD_SERVICEACCOUNT_KEY
+   - The name of the Google Cloud project exported in the environment as GCLOUD_PROJECT
+*/
+
+var targetAPIVersion = schema.GroupVersion{
+	Group:   "targets.triggermesh.io",
+	Version: "v1alpha1",
+}
+
+const (
+	targetKind     = "GoogleCloudStorageTarget"
+	targetResource = "googlecloudstoragetargets"
+)
+
+const gcpCredentialsJosnSecretKey = "credentialsJson"
+
+var _ = Describe("Google Cloud Storage target", func() {
+	// NOTE: bucket names aren't allowed to contain "goog"
+	f := framework.New("gcloudstoragetarget")
+
+	var ns string
+
+	var trgtClient dynamic.ResourceInterface
+
+	var bucketName string
+	var gcpSecret *corev1.Secret
+
+	BeforeEach(func() {
+		ns = f.UniqueName
+
+		gvr := targetAPIVersion.WithResource(targetResource)
+		trgtClient = f.DynamicClient.Resource(gvr).Namespace(ns)
+	})
+
+	Context("a target is deployed", func() {
+		var trgtURL *url.URL
+		var storageClient *storage.Client
+
+		var sentEvent *cloudevents.Event
+
+		var serviceaccountKey string
```

Comment:
This variable is not used outside of the `BeforeEach` just below, so it doesn't need to be declared inside the Ginkgo spec's scope.